iceoryx_binding_c/test/moduletests/test_log.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_utils/log/logmanager.hpp"
+
+using namespace iox::log;
+
+extern "C" {
+#include "iceoryx_binding_c/log.h"
+}
+
+#include "test.hpp"
+
+using namespace ::testing;
+
+TEST(iox_log_test, LogLevelIsSetCorrectly)
+{
+    auto& logManager = iox::log::LogManager::GetLogManager();
+
+    iox_set_loglevel(Iceoryx_LogLevel_Off);
+    EXPECT_EQ(logManager.DefaultLogLevel(), LogLevel::kOff);
```

Comment:
Test via C++ API (it is essentially just an enum translation in the "implementation"). I think either the C++ method has a misleading name or I could not find the correct method: I miss something properly named to get the current log level, but ``DefaultLogLevel()`` seems to do the trick.

Reply:
`DefaultLogLevel` is the log level which is used for new Logger. The current logger implementation is inspired by Autosar and they do not have a singleton approach but one has to create a logger before using it. As library this is not nice to use, therefore we have a singleton in `iceoryx_posh` and which uses `DefaultLogLevel` when the singleton is created. This also means that `iox_set_loglevel` must be called before anything eslse

Reply:
Yes I added a note at the API. It might not be ideal but it is functional for now and the best we can do without changing the logger concept (no time for this).